synapse/storage/databases/main/room.py

```diff
@@ -1388,6 +1467,9 @@ async def maybe_store_room_on_outlier_membership(
         # mark the room as having an auth chain cover index.
         has_auth_chain_index = await self.has_auth_chain_index(room_id)
 
+        create_event = await self.get_create_event_for_room(room_id)
+        room_creator = create_event.content.get("creator", None)
```

Comment:
I'm not so sure about this case being totally right. The MSC2716 stuff (Complement test suite) doesn't touch this code at all to be sure. And there are many usages of this function to check out.

I don't know if all of the usages of `maybe_store_room_on_outlier_membership`  have room state event ID's in place when we call it to rely on `get_create_event_for_room(room_id)`

For example with the `upsert_room_on_join` case above, I had to change it to pass in the `auth_events` because no state events were persisted at that time to use `get_create_event_for_room` which derives from them. But `upsert_room_on_join` also only has one usage so it's easy to reason about and refactor.

Reply:
> **I think what this really comes down to is when a user gets a federated invitation to the room, does the homeserver start receiving messages in the room before the user joins (while in the invited state)?** Our goal is to process MSC2716 related events consistently across homeservers which are restricted to the room creator.

@richvdh Any insight here?

Reply:
> I think what this really comes down to is when a user gets a federated invitation to the room, does the homeserver start receiving messages in the room before the user joins (while in the invited state)? 

No, the homeserver should not receive any regular events in a room until the user has properly joined. You can see this at https://github.com/matrix-org/synapse/blob/develop/synapse/handlers/federation_event.py#L223 - any incoming events for rooms we are not in get dropped. The exception to that is "out-of-band" membership events - such as further invite events.

So, yes, I think you're safe not knowing about the creator while you're in the "invited" state.

Reply:
Thanks for for the info and source 😀 glad we don't have to go down that rabbit hole of hooking everything up there.